mbed_flasher/flashers/FlasherMbed.py

```diff
@@ -160,18 +162,29 @@ def flash(self, source, target, method, no_reset):
                 return EXIT_CODE_FILE_COULD_NOT_BE_READ
 
             self.logger.debug("copy finished")
-            sleep(4)
 
-            new_target = MountVerifier(self.logger).check_points_unchanged(target)
-
-            if isinstance(new_target, int):
-                return new_target
-
-            thread = Thread(target=self.runner,
-                            args=([target['mount_point'], tail],))
-            thread.start()
-            while thread.is_alive():
-                thread.join(2.5)
+            try:
+                # expected behaviour after file copied for mount point is to disappear
+                self._check_mount_point_disappear(mount_point)
+                self.logger.info("mount point disappear success")
+            except MountPointDisappearTimeoutError as err:
+                # The pylint warnings: lost-exception happens if some error other than the
+                # MountPointDisappearTimeoutError occurs in the above try block, it has chance
+                # to be overwritten by the following finally block raised errors
+                self.logger.error(err)
+                # MountVerifier(self.logger).check_points_unchanged(target)
+                return EXIT_CODE_MOUNT_POINT_DISAPPEAR_TIMEOUT
+            finally:
```

Comment:
Why is this finally block here?
If mount point doesn't disappear why would the finally block need to be run?

Reply:
not needed anymore